### PR TITLE
[network] Network application senders now wrap `NetworkSender`

### PR DESCRIPTION
**Pull request by phlip9**

### CODE REVIEW

File: network/src/validator_network/state_synchronizer.rs [hunk 1]
```diff
@@ -23,32 +25,33 @@ pub const STATE_SYNCHRONIZER_MSG_PROTOCOL: &[u8] = b"/libra/state-synchronizer/d
 /// `channel::Receiver<NetworkNotification>`.
 pub type StateSynchronizerEvents = NetworkEvents<StateSynchronizerMsg>;
 
+/// The interface from StateSynchronizer to Networking layer.
+///
+/// This is a thin wrapper around an `channel::Sender<NetworkRequest>`, so it is
```
Comment (bothra90) [inline network/src/validator_network/state_synchronizer.rs]: comment needs to be updated

Comment (phlip9) [reply network/src/validator_network/state_synchronizer.rs]: Ahh this was intended. I wanted to signal that this was in fact a wrapper around a `channel::Sender<NetworkRequest>` despite the `NetworkSender` indirection.

Comment (bothra90) [reply network/src/validator_network/state_synchronizer.rs]: Maybe just mention in the indirection in the comment? ..."This is a thin wrapper around NetworkSender, which in turn wraps channel::Sender<...?" for example.

**Pull request closed without merging**